### PR TITLE
fix: CYS - change heading color

### LINKED ISSUE #42940: [CYS] Color palette - Create your own
Could we please investigate why I can't change the color for the Heading in the Create your own functionality (Change your color palette section)? Thank you!  

image

### PR DESCRIPTION
**Pull request by gigitux**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

Even if this PR should work, I didn't still get _why_ these changes are necessary :/. 

Maybe @chihsuan has a better context.

I don't understand why:
- we have to reset the core/heading color. TT4 doesn't have any customization for the core/heading block. ([theme.json](https://github.com/WordPress/twentytwentyfour/blob/trunk/theme.json))
- without this fix, the heading color is `--wp--preset--color--contrast`. The TT4 does not define this variable. Why does it happen? 

 

Closes #42940.

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Make sure you have WooCommerce Beta Tester installed.
2. Enable the customize-store feature flag in Tools -> WCA Test Helper -> Features.
3. Go to WooCommerce -> Home and click on Customize your store task.
4. Set up the store.
5. In the sidebar, click on `change the color palette`.
6. In the section `create your own palette`, update the heading color.
7. Ensure that the heading elements have the updated color.
8. Test the predefined palette.
9. Ensure that the heading elements have the updated color.

### Changelog entry

-   [x] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [ ] Patch
-   [x] Minor
-   [ ] Major

#### Type

-   [x] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message 
Fix: CYS - change heading color.
#### Comment 

</details>

### CODE REVIEW

**Review by chihsuan — APPROVED** [review]
Thanks for fixing this! LGTM and tested well. 🚢

**Comment by chihsuan** [comment]
> we have to reset the core/heading color. TT4 doesn't have any customization for the core/heading block. ([theme.json](https://github.com/WordPress/twentytwentyfour/blob/trunk/theme.json))

Because when changing the heading color via “create your own”, the _style parameter callback from GB’s StylesColorPanel component [here](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce-admin/client/customize-store/assembler-hub/sidebar/global-styles/color-panel.jsx#L29) sets “the heading” to a fixed color hex code but not “core/heading”.  We reset “core/heading” to make sure the heading and “core/heading” use the same color since when the “core/heading” color is not set, the fallback is to the “heading” color.

> without this fix, the heading color is --wp--preset--color--contrast. The TT4 does not define this variable. Why does it happen?

I’m not sure how exactly --wp--preset--color--contrast is set but we need to set “heading” to var(--wp--preset--color--foreground) because our color palettes don’t define --wp--preset--color--contrast value and we already set core/heading color to var(--wp--preset--color--foreground) [here](https://github.com/woocommerce/woocommerce/blob/a9550d8894ea5de1f02fc954ff41db8eac487f73/plugins/woocommerce-admin/client/customize-store/assembler-hub/sidebar/global-styles/color-palette-variations/constants.ts#L1438-L1449).
In short, we need to ensure the heading and “core/heading” styles are consistent.

p1703246494116929/1703169811.055979-slack-C053716F2H2

**Review by nefeline — APPROVED** [review]

**Comment by nefeline** [comment]
Thanks for your work here, @gigitux, and also to @chihsuan for providing more context regarding the custom color palette.
